internal/olm/operator/bundle/install.go

```diff
@@ -98,3 +185,137 @@ func (i *Install) setup(ctx context.Context) error {
+		imageDeclConfig.Bundles = append(imageDeclConfig.Bundles, bundleDeclConfig.Bundles[0])
+		imageDeclConfig.Channels = append(imageDeclConfig.Channels, bundleDeclConfig.Channels[0])
+		imageDeclConfig.Packages = append(imageDeclConfig.Packages, bundleDeclConfig.Packages[0])
+
+		if len(bundleDeclConfig.Others) > 0 {
+			imageDeclConfig.Others = append(imageDeclConfig.Others, bundleDeclConfig.Others[0])
+		}
+
+		log.Infof("Inserted the new bundle %q into the index image: %s", bundleDeclConfig.Bundles[0].Name, indexImage)
+	}
+
+	return imageDeclConfig, nil
+}
+
+// createFBC generates an FBC by creating bundle, package and channel blobs.
+func (f *FBCContext) createFBC(ctx context.Context) (*declarativeconfig.DeclarativeConfig, error) {
+	var (
+		declcfg        *declarativeconfig.DeclarativeConfig
+		declcfgpackage *declarativeconfig.Package
+		err            error
+	)
+
+	// Rendering the bundle image into a declarative config format.
+	log.SetOutput(ioutil.Discard)
+	render := action.Render{
+		Refs: f.Refs,
+	}
+
+	// generate bundles by rendering the bundle objects.
+	declcfg, err = render.Run(ctx)
+	if err != nil {
+		log.Errorf("Error in rendering the bundle image: %v", err)
+		return nil, err
+	}
+	log.SetOutput(os.Stdout)
+
+	// Ensuring a valid bundle size.
+	if len(declcfg.Bundles) < 0 {
+		return nil, fmt.Errorf("bundles rendered are less than 0 for the bundle image")
+	}
+
+	// init packages.
+	init := action.Init{
+		Package:        f.Package,
+		DefaultChannel: f.ChannelName,
+	}
+
+	// generate packages.
+	declcfgpackage, err = init.Run()
+	if err != nil {
+		log.Errorf("Error in generating packages for the FBC: %v", err)
+		return nil, err
+	}
+	declcfg.Packages = []declarativeconfig.Package{*declcfgpackage}
+
+	// generate channels.
+	channel := declarativeconfig.Channel{
+		Schema:  schemaChannel,
+		Name:    f.ChannelName,
+		Package: f.Package,
+		Entries: []declarativeconfig.ChannelEntry{f.ChannelEntry},
+	}
+
+	declcfg.Channels = []declarativeconfig.Channel{channel}
+
+	return declcfg, nil
+}
+
+// stringifyDecConfig writes the generated declarative config to a string.
+func stringifyDecConfig(declcfg *declarativeconfig.DeclarativeConfig) (string, error) {
+	var buf bytes.Buffer
+	err := declarativeconfig.WriteJSON(*declcfg, &buf)
+	if err != nil {
+		log.Errorf("Error writing to JSON encoder: %v", err)
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+// validateFBC converts the generated declarative config to a model and validates it.
+func validateFBC(declcfg *declarativeconfig.DeclarativeConfig) error {
+	// convert declarative config to model
+	FBCmodel, err := declarativeconfig.ConvertToModel(*declcfg)
+	if err != nil {
+		log.Errorf("Error converting the declarative config to model: %v", err)
+		return err
+	}
+
+	if err = FBCmodel.Validate(); err != nil {
+		log.Errorf("Error validating the generated FBC: %v", err)
+		return err
+	}
```

Comment:
There's no need to call `Validate` after the conversion ([`ConvertToModel` validates internally](https://github.com/operator-framework/operator-registry/blob/9e59597a435341d857283011c8ede1c16e5e292b/alpha/declcfg/declcfg_to_model.go#L167-L169)).

Reply:
Didn't realize it was validating it internally. Got rid of the `Validate()` function.